### PR TITLE
[DOCS] Release highlights for 7.16.0

### PR DESCRIPTION
**Pull request by danhermann**

### CODE REVIEW

File: docs/reference/release-notes/highlights.asciidoc [hunk 1]
```diff
@@ -30,6 +30,36 @@ Other versions:
...
+
+=== SQL: Support for cross-cluster search
+
+Cross-cluster search (CCS) is now supported in SQL using the CCS-specific
+`cluster:index` notation where `cluster` maps to a SQL catalog and
+`index` to a table.
+
+=== Search: Node-level can-match action
+
+
```
Comment (jrodewig) [inline docs/reference/release-notes/highlights.asciidoc:42]: @ywelsch Do you mind helping provide a draft for this section?

I took a look at https://github.com/elastic/elasticsearch/pull/78765, but I wasn't familiar with the `can-match` action. If we include some more introductory information about that action and when ES may use it, I think we can use the rest of the description in the PR.

File: docs/reference/release-notes/highlights.asciidoc [hunk 2]
```diff
@@ -30,6 +30,39 @@ Other versions:
...
+Cross-cluster search (CCS) is now supported in SQL using the CCS-specific
+`cluster:index` notation where `cluster` maps to a SQL catalog and
+`index` to a table.
+
+=== Search: Can-match action performed per-node rather than per-shard
+
+Previously, the can-match phase issued individual requests for each shard
+in a search request resulting in high overhead when many shards were
+targeted. Now these requests are grouped by node so that no more
+than one request per node is sent.
```
Comment (jrodewig) [inline docs/reference/release-notes/highlights.asciidoc:45]: This one is a little tough. I think we need to explain can-match for this to make sense, but we don't do that anywhere currently in the docs. I attempt to do that with this suggestion.

```suggestion
=== Search: Improved can-match phase for scalability

If a search would hit a large number of shards, the search process includes a
pre-filter phase, also called the can-match phase. During this phase, {es}
checks if an impacted shard actually contains data that could potentially match
the search query. If not, {es} doesn't run the query on the shard. For example,
if a query includes a `@timestamp` date range filter, {es} will use the
can-match phase to exclude shards that have no data within the date range.

Previously, the search's coordinating node sent an individual request to each
shard checked during the can-match phase. However, if the search needed to check
thousands of shards, the coordinating node would need to handle thousands of
requests and responses, resulting in high overhead.

With 7.16, the coordinating node instead sends a single request to each data
node during the can-match phase. This request covers checks for all impacted
shards on the node. This significantly reduces the number of requests and lowers
related overhead.
```

File: docs/reference/release-notes/highlights.asciidoc [hunk 3]
```diff
@@ -38,4 +71,13 @@ succession. Rather than type the same event criteria multiple times, you can use
...
 For more details, check out the
 {ref}/eql-syntax.html#eql-with-runs-statement[EQL syntax documentation].
+
+
+=== Field capabilities: Results gathered by node rather than index
+
+Previously, field caps were gathered by sending a request per index to the
+appropriate nodes in the cluster resulting in high overhead when many indices
+were targeted. Now requests that target indices on a single node are grouped
+together resulting in no more than a single request per node.
```
Comment (jrodewig) [inline docs/reference/release-notes/highlights.asciidoc:81]: I'd consider combining this with the "can-match" section or moving it side-by-side. The benefits and takeaways are largely the same.

**Review by jrodewig — COMMENTED** [review]
Thanks @danhermann. I left some suggestions.

File: docs/reference/release-notes/highlights.asciidoc (hunk 3 above)
Comment (danhermann) [reply docs/reference/release-notes/highlights.asciidoc]: Thanks, @jrodewig. I've made those changes.

**Review by jrodewig — APPROVED** [review]
I pushed https://github.com/elastic/elasticsearch/pull/81370/commits/b7faf52b2bf1a6c4cb9bf8d8ba69829bd45a55ad and https://github.com/elastic/elasticsearch/pull/81370/commits/cfa116921be3be9dc42c8a8ecf499ecbdb1b8e78 to add some needed `discrete` tags and clean up some of my typos/awkward wording.

If that looks okay to you, this looks good to me. Thanks!

**Pull request merged**